### DISCUSSION TITLE
How to be autenticated in the API

### DISCUSSION THREAD

**Discussion by sbrunner** [jonaswinkler/paperless-ng#1413] [Support] [answered]
Hello, I try to do some PUT using the API with simple auth, ad it's written here it should work:
https://paperless-ng.readthedocs.io/en/latest/api.html#authorization

But I get the following result (cleaned):
```
Forbidden (403)

CSRF verification failed. Request aborted.
You are seeing this message because this site requires a CSRF cookie when submitting forms. This cookie is required for security reasons, to ensure that your browser is not being hijacked by third parties.
If you have configured your browser to disable cookies, please re-enable them, at least for this site, or for \xe2\x80\x9csame-origin\xe2\x80\x9d requests.

More information is available with DEBUG=True.
```

Is it wanted that this CSRF cookie is enabled when we want to use the REST API, or how to correctly manage that?

On `/api/token/` I get same kind of result:
```json
{
  "detail":"CSRF Failed: CSRF cookie not set."
}
```

**Comment by TobenderZephyr** [comment]
I just tested it with YARC just with Authorization Headers and had no problems at all.
paperless-ng inside docker container -> traefik as reverse proxy.

Are you running any kind of Reverse Proxy which is unable to handle it or did you try directly on the container?

**Reply by sbrunner** [reply]
I didn't have any reverse proxy, finally I succeed using this code:
https://gist.github.com/sbrunner/8aa5b83141ea01b75e6a0f37637c3202

**Comment by sbrunner** [comment, marked as answer]
Solution: https://gist.github.com/sbrunner/8aa5b83141ea01b75e6a0f37637c3202
